pandas/core/indexes/multi.py

```diff
@@ -1575,7 +1594,7 @@ def dropna(self, how="any"):
             raise ValueError(f"invalid how option: {how}")
 
         new_codes = [level_codes[~indexer] for level_codes in self.codes]
-        return self.copy(codes=new_codes, deep=True)
+        return self.copy(deep=True).set_codes(codes=new_codes)
```

Comment:
isn't this copying twice?

Reply:
Looking again, the call to `.copy` isn't needed because a `MultiIndex` is immutable, so I've removed that call.